core/src/com/unciv/ui/EmpireOverviewScreen.kt

```diff
@@ -324,25 +324,41 @@ class EmpireOverviewScreen(val viewingPlayer:CivilizationInfo) : CameraStageBase
     }
 
 
-    fun playerKnows(civ:CivilizationInfo) = civ==viewingPlayer ||
+    private fun playerKnows(civ:CivilizationInfo) = civ==viewingPlayer ||
             viewingPlayer.diplomacy.containsKey(civ.civName)
 
-    fun getDiplomacyGroup(): Group {
+    private fun getDiplomacyGroup(): Group {
         val relevantCivs = viewingPlayer.gameInfo.civilizations.filter { !it.isBarbarian() && !it.isCityState() }
         val freeWidth = stage.width
         val freeHeight = stage.height - topTable.height
-        val groupSize = if (freeWidth > freeHeight) freeHeight else freeWidth
         val group = Group()
-        group.setSize(groupSize,groupSize)
+        group.setSize(freeWidth,freeHeight)
         val civGroups = HashMap<String, Actor>()
+        val civLines = HashMap<String, MutableSet<Actor>>()
         for(i in 0..relevantCivs.lastIndex){
             val civ = relevantCivs[i]
 
             val civGroup = getCivGroup(civ, "", viewingPlayer)
 
             val vector = HexMath.getVectorForAngle(2 * Math.PI.toFloat() *i / relevantCivs.size)
             civGroup.center(group)
-            civGroup.moveBy(vector.x*groupSize/3, vector.y*groupSize/3)
+            civGroup.moveBy(vector.x*freeWidth/3, vector.y*freeHeight/3)
+            civGroup.onClick {
+
+                val numberOfDisabled = civLines.values.count { it.firstOrNull()?.isVisible == false }
+                if (numberOfDisabled == 0)
+                    // all enabled: disable all except current one
+                    civLines.forEach{ (s, set) -> set.forEach {it.isVisible = s == civ.civName} }
+                else {
+                    val chosenOn = civLines[civ.civName]?.firstOrNull()?.isVisible
```

Comment:
Again, meaningful name. areCivLinesDisplayed, or isCivEnabled
If we've chosed 'enabled/disabled' as our naming scheme for 'lines of civ are visible' let's stick to it =)